src/all/pixiv/src/eu/kanade/tachiyomi/extension/all/pixiv/Pixiv.kt

```diff
@@ -85,6 +88,25 @@ class Pixiv(override val lang: String) : HttpSource() {
     private lateinit var searchIterator: Iterator<SManga>
 
     override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        // Deeplink selection of specific IDs: simply fetch the single object and return
+        if (query.startsWith(ID_ILLUST_PREFIX) &&
+            query.removePrefix(ID_ILLUST_PREFIX).matches("\\d+".toRegex())
+        ) {
```

Comment:
```suggestion
        if (query.startsWith(ID_ILLUST_PREFIX)) {
```
no need to check, since anything else will result in 404 or similar from pixiv

Reply:
The reason I added that check is so regular queries that just happen to look like `sid:dharta or sth` aren't captured by this mechanism. Idea being that the user either knows about this special handling, or must be *extremely* unlucky to trigger it. But if you don't think it's important to go that far, yeah we can skip the check.

Reply:
I've decided (for now) to move completely to the URL-based system, meaning this isn't really relevant anymore. I've left the code for parsing and handling the special search queries in, including this check, because especially the way they're now conceptually set up there needs to be *some* validation. But it's currently not doing anything anyway, and the exact way it's gonna be used/useful/changed/deleted in the future is completely up in the air for now.